### PR TITLE
Add CSS Grid test for old & new syntaxes

### PR DESCRIPTION
**Pull request by KuraFire**
While CSS Grid gets rolled out, it’s useful to have a way to target
IE11 which does not support @supports() but does support old CSS Grid
syntax, as does the new IE14.

### CODE REVIEW

**Comment by ryanseddon** [comment]
Can you add this to [config-all.json](https://github.com/Modernizr/Modernizr/blob/master/lib/config-all.json) too please

**Comment by ryanseddon** [comment]
Normally we have each test separate but I think this makes sense to put it in one since most people will want both.

File: feature-detects/css/cssgrid.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "property": ["cssgrid", "cssgridlegacy"],
+  "tags": ["css"],
+  "notes": [{
+    "name": "The new, standardized CSS Grid",
+    "href": "https://www.w3.org/TR/css3-grid-layout/"
+  }, {
+    "name": "The _old_ CSS Grid",
+    "href": "https://www.w3.org/TR/2011/WD-css3-grid-layout-20110407/"
+  }],
+  "warnings": ["`grid-columns` is only in the old syntax, `grid-column` exists in both and so `grid-template-rows` is used for the new syntax."]
```
Comment (patrickkettner) [inline feature-detects/css/cssgrid.js]: this isn't really a warning. Warnings should be about the detect themselves (i.e. 🚨 giamundo test🚨 , or listing possible edge case failures. Since the end user of modernizr won't need this information, but modernizr developers would, could you move this to a straight up JS comment on line 17?

**Comment by patrickkettner** [comment]
lgtm
